src/modules/file/libkvifile.cpp

```diff
@@ -1478,19 +1478,25 @@ static bool file_kvs_fnc_diskSpace(KviKvsModuleFunctionCall * c)
 	if(szPath.isEmpty())
 		szPath = ".";
 
-	const char * pcPath = szPath.toUtf8().data();
 	long long int fTotal;
 	long long int fFree;
 // this for win
 #if defined(COMPILE_ON_WINDOWS) || defined(COMPILE_ON_MINGW)
 	ULARGE_INTEGER free, total;
-	GetDiskFreeSpaceExA((LPCTSTR)pcPath, NULL, &total, &free);
+	if (GetDiskFreeSpaceEx(szPath.toStdWString().c_str(), NULL, &total, &free) == 0) {
+		c->warning(__tr2qs("An error occurred retrieving the amount of free space in '%Q'"), &szPath);
+		return true;
+	}
 	fFree = free.QuadPart;
 	fTotal = total.QuadPart;
 #else
 	// this one for linux and macos
+	const char * pcPath = szPath.toUtf8().data();
```

Comment:
`QString::toUtf8` returns a temporary `QByteArray`, so `pcPath` is pointing to invalid locations in memory.  Consider `auto path = szPath.toUtf8();` and then using `path.data()`.